quantecon/lqcontrol.py

```diff
@@ -327,3 +329,279 @@ def compute_sequence(self, x0, ts_length=None, method='doubling',
+
+    .. math::
+
+         r(x_t, s_t, u_t) :=
+            (x_t' R(s_t) x_t + u_t' Q(s_t) u_t + 2 u_t' N(s_t) x_t)
+
+    subject to the law of motion
+
+    .. math::
+
+         x_{t+1} = A(s_t) x_t + B(s_t) u_t + C(s_t) w_{t+1}
+
+    Here :math:`x` is n x 1, :math:`u` is k x 1, :math:`w` is j x 1 and the
+    matrices are conformable for these dimensions.  The sequence :math:`{w_t}`
+    is assumed to be white noise, with zero mean and
+    :math:`\mathbb{E} [ w_t' w_t ] = I`, the j x j identity.
+
+    If :math:`C` is not supplied as a parameter, the model is assumed to be
+    deterministic (and :math:`C` is set to a zero matrix of appropriate
+    dimension).
+
+    The optiaml value function :math:`V(x_t, s_t)` takes the form
+
+    .. math::
+
+         x_t' P(s_t) x_t + d(s_t)
+
+    and the optimal policy is of the form :math:`u_t = -F(s_t) x_t`.
+
+    Parameters
+    ----------
+    Π : array_like(float, ndim=2)
+        The Markov chain transition matrix with dimension m x m.
+    Qs : array_like(float)
+        Consists of m symmetric and non-negative definite payoff
+        matrices Q(s) with dimension k x k that corresponds with
+        the control variable u for each Markov state s
+    Rs : array_like(float)
+        Consists of m symmetric and non-negative definite payoff
+        matrices R(s) with dimension n x n that corresponds with
+        the state variable x for each Markov state s
+    As : array_like(float)
+        Consists of m state transition matrices A(s) with dimension
+        n x n for each Markov state s
+    Bs : array_like(float)
+        Consists of m state transition matrices B(s) with dimension
+        n x k for each Markov state s
+    Cs : array_like(float), optional(default=None)
+        Consists of m state transition matrices C(s) with dimension
+        n x j for each Markov state s. If the model is deterministic
+        then Cs should take default value of None
+    Ns : array_like(float), optional(default=None)
+        Consists of m cross product term matrices N(s) with dimension
+        k x n for each Markov state,
+    beta : scalar(float), optional(default=1)
+        beta is the discount parameter
+
+    Attributes
+    ----------
+    Π, Qs, Rs, Ns, As, Bs, Cs, beta : see Parameters
+    Ps : array_like(float)
+        Ps is part of the value function representation of
+        :math:`V(x, s) = x' P(s) x + d(s)`
+    ds : array_like(float)
+        ds is part of the value function representation of
+        :math:`V(x, s) = x' P(s) x + d(s)`
+    Fs : array_like(float)
+        Fs is the policy rule that determines the choice of control in
+        each period at each Markov state
+    m : scalar(int)
+        The number of Markov states
+    k, n, j : scalar(int)
+        The dimensions of the matrices as presented above
+
+    """
+
+    def __init__(self, Π, Qs, Rs, As, Bs, Cs=None, Ns=None, beta=1):
+
+        # == Make sure all matrices can be treated as 3D arrays == #
```

Comment:
2D?

Reply:
Here I was trying to see that all the arrays (`Qs`, `Rs`, etc.) consist of 2d arrays, but I guess it is confusing. Would "Make sure all matrices for each state are 2D arrays" be a little better?

Reply:
No, I just mean this: In your comment on line 423, you say `3D` but do you actually intend to say `2D`?

Reply:
I actually intended to say 3D since I was referring to the shape of `Qs` array (and `Rs` etc.) and `Qs` has shape m (number of Markov states) x n (number of state variables) x n, but after you pointed this out, I realized `Qs` is not a matrix and saying "matrices are 3D arrays" does not make any sense.. so I think maybe the comment I proposed above would be a little more clear